packages/node/src/indexer/benchmark.service.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020-2021 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { OnEvent } from '@nestjs/event-emitter';
+import * as moment from 'moment';
+import 'moment-duration-format';
+import { MetricPayload } from '../prometheus/types';
+import { getLogger } from '../utils/logger';
+import { delay } from '../utils/promise';
+import { BlockPayload } from './types';
+
+const SAMPLING_TIME_VARIANCE = 15;
+const logger = getLogger('benchmark');
+
+export class BenchmarkService {
+  private currentProcessingHeight: number;
+  private currentProcessingTimestamp: number;
+  private targetHeight: number;
+  private lastRegisteredHeight: number;
+  private lastRegisteredTimestamp: number;
+  private blockPerSecond: number;
+  private needReset = true;
```

Comment:
why need this flag? under which circumstance we don't want to reset last height and last timestamp?